ports/stm32/machine_pixelbitstream.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/mphal.h"
+#include "modmachine.h"
+
+/*
+Example usage:
+
+import machine
+
+class NeoPixel:
+    ORDER = (1, 0, 2, 3)
+    TIMING = (350, 800, 700, 600)
```

Comment:
I would label this as WS2812S_TIMING, perhaps include other timings, from here:
https://github.com/micropython/micropython/blob/b635d0eb17fd3c0d4a94b9f261c11a7c461cb5d4/ports/esp32/espneopixel.c

You also likely want to support the is800khz parameter from the ESP32 port:
http://docs.micropython.org/en/latest/esp32/quickref.html#neopixel-driver

Just for the sake of portability. Perhaps a named param? Right now is800khz would get mapped to bpp.

Reply:
Also, there is a final timing parameter, which is the latch time. This should be included.